src/update.c

```diff
@@ -270,6 +271,15 @@ static int main_update()
 				fclose(verfile);
 			}
 
+			/* The .valid-mix file contains the mixin version. When updating
+			 * to a new upstream version, erase this file so that mixin can
+			 * recreate it with a version based on the new upstream. */
+			string_or_die(&mixin_verfile, "%s/%s", MIX_DIR, ".valid-mix");
+			if (remove(mixin_verfile)) {
+				fprintf(stderr, "ERROR: Unable to remove %s\n", mixin_verfile);
+			}
+			free_string(&mixin_verfile);
+
 			if (run_command("/usr/bin/mixin"
```

Comment:
If we have a problem in this line (power goes down, there's any problem in mixin execution, the user ctrl-c swupd) we won't have a .valid-mix file in the system anymore. Isn't this a problem when trying to run swupd again?

Reply:
You're right about power loss and killing swupd causing a problem here. They would have to run mixin build again to re-create the .valid-mix file.